src/Agent.Worker/Build/TrackingConfig.cs

```diff
@@ -65,20 +64,31 @@ public TrackingConfig(IExecutionContext executionContext, ServiceEndpoint endpoi
         [JsonProperty("agent_builddirectory")]
         public string BuildDirectory { get; set; }
 
-        public string CollectionName { get; set; }
+        public string CollectionUrl { get; set; }
 
         public string DefinitionName { get; set; }
 
         [JsonProperty(FileFormatVersionJsonProperty)]
         public int FileFormatVersion
         {
-            get { return 2; }
+            get
+            {
+                return 3;
+            }
 
             set
             {
-                if (value != 2)
+                // Version 3 changes:
+                //   CollectionName was removed.
+                //   CollectionUrl was added.
+                switch (value)
                 {
-                    throw new NotSupportedException();
+                    case 3:
+                    case 2:
+                        break;
+                    default:
+                        // Should never reach here.
+                        throw new NotSupportedException();
```

Comment:
I think throwing from the setter is OK in this case for two reasons:
1) We should never hit this case. This is just a sanity check.
2) The serializer is the only place that calls the setter. So if it were to blow up, it would be on deserialization, not some normal caller.